torch/_dynamo/variables/lists.py

```diff
@@ -25,29 +26,47 @@
 from .functions import UserFunctionVariable, UserMethodVariable
 
 
-def _listlike_contains_helper(items, search, tx, options):
+def _get_fake_tensor(vt):
+    fake_tensor = vt.as_proxy().node.meta.get("example_value")
+    if not is_fake(fake_tensor):
+        unimplemented("Cannot check Tensor object identity without its fake value")
+    return fake_tensor
+
+
+def _listlike_contains_helper(items, search, tx, options, check_tensor_identity=False):
     if search.is_python_constant():
-        result = any(
+        found = any(
             x.is_python_constant()
             and x.as_python_constant() == search.as_python_constant()
             for x in items
         )
-        return variables.ConstantVariable.create(result, **options)
+        return variables.ConstantVariable.create(found, **options)
 
     from .builtin import BuiltinVariable
 
-    result = None
+    must_check_tensor_id = False
+    if check_tensor_identity and isinstance(search, variables.TensorVariable):
+        must_check_tensor_id = True
+        # Match of Tensor means match of FakeTensor
+        search = _get_fake_tensor(search)
+
+    found = None
     for x in items:
-        check = BuiltinVariable(operator.eq).call_function(tx, [x, search], {})
-        if result is None:
-            result = check
+        if must_check_tensor_id:
+            if isinstance(x, variables.TensorVariable):
+                if search is _get_fake_tensor(x):  # Object equivalence
```

Comment:
Actually, do we need to set any object equivalence guard here, or do we assume that these were already set somewhere before?

Reply:
For tensors, we guard both aliasing and non-aliasing.